src/app/navigation/hooks/closeModalHook.ts

```diff
@@ -1,3 +1,11 @@
 import { AppServices, MiddlewareAPI } from '../../types';
 
-export const closeModal = (services: MiddlewareAPI & AppServices) => services.history.goBack;
+export const closeModal = () => (services: MiddlewareAPI & AppServices) => () => {
+  if (services.history.location.state === undefined) {
```

Comment:
It's undefined if the previous history record is not on rex—I'll add a comment. Maybe I'm forgetting some part of this but we only want to send them back if that keeps them on rex, otherwise we clear the query to close the modal.

Reply:
ok yea i remember wanting to replace instead if there is no previous record, but why is it setting search to an empty string? is that just to make it different at all?